### PR TITLE
Depend on ament_cmake_ros to default SHARED to ON

### LINKED ISSUE #97: Shared library build when `BUILD_SHARED_LIBS` is set to `OFF`
## Bug report

**Required Info:**

- Operating System: 
  - Ubuntu 22.04 (Docker)
- Installation type: 
  - source
- Version or commit hash: 
  - humble
- DDS implementation: 
  - Fast-RTPS

#### Steps to reproduce issue

```
FROM docker.io/ubuntu:22.04

ENV DEBIAN_FRONTEND=noninteractive
ENV TZ=Europe/Berlin

RUN apt-get update && apt-get install -y \
	locales \
	software-properties-common \
	curl gnupg lsb-release

RUN locale-gen en_US en_US.UTF-8 &&\
	update-locale LC_ALL=en_US.UTF-8 LANG=en_US.UTF-8 &&\
	add-apt-repository universe
	
RUN curl -sSL https://raw.githubusercontent.com/ros/rosdistro/master/ros.key -o /usr/share/keyrings/ros-archive-keyring.gpg &&\
	echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2/ubuntu $(lsb_release -cs) main" | tee /etc/apt/sources.list.d/ros2.list > /dev/null

RUN apt-get update && apt-get install -y \
	python3-flake8-docstrings \
	python3-pip \
	python3-pytest-cov \
	ros-dev-tools \
	python3-flake8-blind-except \
	python3-flake8-builtins \
	python3-flake8-class-newline \
	python3-flake8-comprehensions \
	python3-flake8-deprecated \
	python3-flake8-import-order \
	python3-flake8-quotes \
	python3-pytest-repeat \
	python3-pytest-rerunfailures

WORKDIR /root/ros2_humble
RUN mkdir -p /root/ros2_humble/src
RUN vcs import --input https://raw.githubusercontent.com/ros2/ros2/humble/ros2.repos src
RUN apt-get update && apt-get upgrade -y && rosdep init && rosdep update &&\
	rosdep install --from-paths src --ignore-src -y --skip-keys "fastcdr rti-connext-dds-6.0.1 urdfdom_headers"

RUN colcon build \
	--cmake-args '-DBUILD_SHARED_LIBS=OFF' \
	--packages-up-to rosidl_typesupport_fastrtps_cpp
```

Build the above Dockerfile.

#### Expected behavior

A static library (`librosidl_typesupport_fastrtps_cpp.a`) should be build when the `BUILD_SHARED_LIBS` flag is set to `OFF`.

#### Actual behavior

A shared library (`librosidl_typesupport_fastrtps_cpp.so`) is build when the `BUILD_SHARED_LIBS` is set to `OFF`.

#### Additional information

The library is apparently build as `SHARED` in https://github.com/ros2/rosidl_typesupport_fastrtps/blob/89b19c15d615fc07479707e6584e29b1f67f5e23/rosidl_typesupport_fastrtps_cpp/CMakeLists.txt#L45
Removing the `SHARED` keyword results in the expected behavior. This change was done in several ros2 projects (see https://github.com/ros2/ros2/issues/306).

### PR DESCRIPTION
**Pull request by tylerjw**
fixes #97

### CODE REVIEW

**Comment by clalancette** [comment]
Same comment as in https://github.com/ros2/geometry2/pull/591#issuecomment-1481542019

**Comment by tylerjw** [comment]
The `package.xml` already has this line. I'm unsure if buildtool_export_depend is the correct type of dependency.

```
<buildtool_export_depend>ament_cmake_ros</buildtool_export_depend>
```

File: rosidl_typesupport_fastrtps_cpp/CMakeLists.txt [hunk 1]
```diff
@@ -21,6 +21,7 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
 endif()
 
 find_package(ament_cmake REQUIRED)
```
Comment (clalancette) [inline rosidl_typesupport_fastrtps_cpp/CMakeLists.txt:23]: Remove this line, it is redundant.

**Review by clalancette — CHANGES_REQUESTED** [review]
Left one fix inline.

> The package.xml already has this line. I'm unsure if buildtool_export_depend is the correct type of dependency.

I think you need it in both.  That is, `buildtool_depend` are the build tools necessary to build *this* package.  `buildtool_export_depend` are build tools that are necessary for downstream packages to depend on and use this package.  So if you update https://github.com/ros2/rosidl_typesupport_fastrtps/blob/166ab452dc93beabc6838a8a233849a24455e298/rosidl_typesupport_fastrtps_cpp/package.xml#L17 to `ament_cmake_ros`, that should do it.

**Comment by tylerjw** [comment]
> Left one fix inline.
> 
> > The package.xml already has this line. I'm unsure if buildtool_export_depend is the correct type of dependency.
> 
> I think you need it in both. That is, `buildtool_depend` are the build tools necessary to build _this_ package. `buildtool_export_depend` are build tools that are necessary for downstream packages to depend on and use this package. So if you update
> 
> https://github.com/ros2/rosidl_typesupport_fastrtps/blob/166ab452dc93beabc6838a8a233849a24455e298/rosidl_typesupport_fastrtps_cpp/package.xml#L17
> 
> to `ament_cmake_ros`, that should do it.

Does this mean my other PR in geometry2 needs to also have a buildtool_export_depend on this?

File: rosidl_typesupport_fastrtps_cpp/CMakeLists.txt (hunk 1 above)
Comment (tylerjw) [reply rosidl_typesupport_fastrtps_cpp/CMakeLists.txt]: I fixed this in a commit I pushed.

**Review by clalancette — APPROVED** [review]

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/18298/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/12824/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/18967/)

**Pull request merged**